webapi/src/test/scala/org/knora/webapi/responders/v2/ontology/CacheSpec.scala

```diff
@@ -0,0 +1,149 @@
+import org.knora.webapi.messages.v2.responder.SuccessResponseV2
+import org.knora.webapi.messages.v2.responder.ontologymessages.ReadOntologyV2
+import org.knora.webapi.settings.KnoraDispatchers
+import org.knora.webapi.sharedtestdata.SharedOntologyTestDataADM
+import org.knora.webapi.store.triplestore.http.HttpTriplestoreConnector
+import org.knora.webapi.util.cache.CacheUtil
+import org.knora.webapi.{IntegrationSpec, TestContainerFuseki}
+
+import java.time.Instant
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+/**
+ * This spec is used to test [[org.knora.webapi.responders.v2.ontology.Cache]].
+ */
+class CacheSpec extends IntegrationSpec(TestContainerFuseki.PortConfig) {
+
+  private implicit val stringFormatter: StringFormatter = StringFormatter.getGeneralInstance
+//  private implicit val ontologyResponder = ResponderManager.
+
+  val additionalTestData = List(
+    RdfDataObject(
+      path = "test_data/ontologies/freetest-onto.ttl",
+      name = "http://www.knora.org/ontology/0001/freetest"
+    ),
+    RdfDataObject(
+      path = "test_data/all_data/freetest-data.ttl",
+      name = "http://www.knora.org/data/0001/freetest"
+    )
+  )
+
+  val defaultFeatureFactoryConfig: FeatureFactoryConfig = new KnoraSettingsFeatureFactoryConfig(settings)
+
+  // start fuseki http connector actor
+  private val fusekiActor = system.actorOf(
+    Props(new HttpTriplestoreConnector()).withDispatcher(KnoraDispatchers.KnoraActorDispatcher),
+    name = "httpTriplestoreConnector"
+  )
+
+  override def beforeAll(): Unit = {
+    CacheUtil.createCaches(settings.caches)
+    waitForReadyTriplestore(fusekiActor)
+    loadTestData(fusekiActor, additionalTestData)
+  }
+
+  "The Ontology Cache" should {
+
+    "successfully load all ontologies" in {
+      // val resFuture = inMemCache.putUserADM(user)
+      // resFuture map { res => res should equal(true) }
+      val ontologiesFromCacheFuture: Future[Map[SmartIri, ReadOntologyV2]] = for {
+        _ <- Cache.loadOntologies(
+               settings,
+               fusekiActor,
+               defaultFeatureFactoryConfig,
+               KnoraSystemInstances.Users.SystemUser
+             )
+        cacheData: Cache.OntologyCacheData       <- Cache.getCacheData
+        ontologies: Map[SmartIri, ReadOntologyV2] = cacheData.ontologies
+      } yield ontologies
+
+      ontologiesFromCacheFuture map { res: Map[SmartIri, ReadOntologyV2] => res.size should equal(13) }
+    }
+
+    "successfully load back all ontologies from cache" in {
+      val ontologiesFromCacheFuture: Future[Map[SmartIri, ReadOntologyV2]] = for {
+        cacheData: Cache.OntologyCacheData       <- Cache.getCacheData
+        ontologies: Map[SmartIri, ReadOntologyV2] = cacheData.ontologies
+      } yield ontologies
+
+      ontologiesFromCacheFuture map { ontologies: Map[SmartIri, ReadOntologyV2] =>
```

Comment:
is this running correctly or do we need asyncSpec?

Reply:
we checked, it seems to work